src/views/virtualmachines/list/VirtualMachinesList.tsx

```diff
@@ -153,46 +153,48 @@ const VirtualMachinesList: FC<VirtualMachinesListProps> = ({ kind, namespace })
       </ListPageHeader>
       <ListPageBody>
         <div className="list-managment-group">
-          <ListPageFilter
-            columnLayout={{
-              columns: columns?.map(({ additional, id, title }) => ({
-                additional,
-                id,
-                title,
-              })),
-              id: VirtualMachineModelRef,
-              selectedColumns: new Set(activeColumns?.map((col) => col?.id)),
-              type: t('VirtualMachine'),
-            }}
-            onFilterChange={(...args) => {
-              onFilterChange(...args);
-              setPagination((prevPagination) => ({
-                ...prevPagination,
-                endIndex: prevPagination?.perPage,
-                page: 1,
-                startIndex: 0,
-              }));
-            }}
-            data={unfilteredData}
-            loaded={loaded}
-            rowFilters={filters}
-            searchFilters={searchFilters}
-          />
           {!isEmpty(vms) && (
```

Comment:
istead of doint !isEmpty u can take the code from line 200 
{loaded && isEmpty(data) && (
          <VirtualMachineEmptyState catalogURL={catalogURL} namespace={namespace} />
        )}
and do a 
if (!loaded) return <Loader />
if (loaded && isEmpty(data)) {
      return    <VirtualMachineEmptyState catalogURL={catalogURL} namespace={namespace} />
        }
above the return of this component.

then we can get rid of all those && checks

Reply:
same goes to templates

Reply:
I don't know. I can see here the same pattern and same situation as for templates. Hence I think we cannot get rid of those checks. Only if we decided to render those components in different situations than we render now. Or maybe I just don't understand what you've suggested?